Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -847,6 +847,48 @@ extension BrowserViewController: BrowserToolbarDelegate {
     }
 }
 
+protocol WindowCloseHelperDelegate: class {
```

Comment:
You were probably planning on it already, but make sure to move this and the class below to a `WindowCloseHelper.swift`.